src/base/resource_client.ts

```diff
@@ -16,17 +16,10 @@ const MAX_WAIT_FOR_FINISH = 999999;
 
 /**
  * Resource client.
- *
- * @param {ApiClientOptions} options
  * @private
  */
-class ResourceClient extends ApiClient {
-    /**
-     * @param {object} [options]
-     * @return {Promise<?object>}
-     * @private
-     */
-    async _get(options = {}) {
+export class ResourceClient extends ApiClient {
+    protected async _get<T extends Record<string, unknown>, R extends unknown>(options: T = {} as T): Promise<R | undefined> {
```

Comment:
does it actually make any difference to define `R extends unknown`? `unknown` is a top type, every type is its extension, so it feels redundant

Reply:
oh yeah, good point. Can simplify it to just `R` 👍